hercules/python_simulators/solar_pysam.py

```
@@ -71,21 +69,33 @@ def __init__(self, input_dict, dt):
             print("self.tz = ", self.tz)
 
         self.needed_inputs = {}
-        self.data = data
+        self.data = weather_data_array
         self.dt = dt
 
         # Save the initial condition
         self.power_mw = input_dict["initial_conditions"]["power"]
         self.dc_power_mw = input_dict["initial_conditions"]["power"]
         self.dni = input_dict["initial_conditions"]["dni"]
         self.aoi = 0
-        self.target_system_capacity = input_dict["target_system_capacity"]
-        self.target_dc_ac_ratio = input_dict["target_dc_ac_ratio"]
+
+        # create pysam model here so that it is not created each timestep
+        system_model = pvsam.new()
+        system_model.AdjustmentFactors.adjust_constant = 0
+        system_model.AdjustmentFactors.dc_adjust_constant = 0
+
+        for k, v in self.model_params.items():
+            try:
+                system_model.value(k, v)
+            except Exception:
+                print(k)
```

Do we not want to raise the exception here?

---

addressed!

---

great!!